azure-iot-device/tests/common/pipeline/test_pipeline_stages_base.py

```diff
@@ -26,6 +26,17 @@
 
 this_module = sys.modules[__name__]
 
+
+# This fixture makes it look like all test in this file  tests are running
+# inside the pipeline thread.  Because this is an autouse fixture, we
+# manually add it to the individual test.py files that need it.  If,
+# instead, we had added it to some conftest.py, it would be applied to
+# every tests in every file and we don't want that.
+@pytest.fixture(autouse=True)
+def apply_fake_pipeline_thread(fake_pipeline_thread):
```

Comment:
I would also want to see tests that show the various methods can ONLY be run in the pipeline thread as well though